scripts/out

```diff
@@ -84,6 +84,8 @@ if "name" in config["params"]:
     name = config["params"]["name"]
 else:
     name = os.environ.get("BUILD_NAME", 'One-off build #{0}'.format(os.environ['BUILD_ID']))
+    if os.environ['BUILD_JOB_NAME']:
```

Comment:
This will raise a `KeyError` if the variable is not defined. It would be better to use `os.getenv('BUILD_JOB_NAME')` which will return `None` in such case.

Reply:
Two comments:

1. This file uses `"` for string quotes, so for uniformity we should use the same.
2. If you look at line 81, env `BUILD_JOB_NAME` will NOT be defined for one-off jobs, so as Olivier points out we need something more robust...

Reply:
I'd suggest changing it to `if 'BUILD_JOB_NAME' in os.environ:` for the sake of uniformity, unless that poses a problem, too. For the string quotes, I guess I should really do a pass over all files and change them to single quotes everywhere.

Reply:
I think it would be cleaner to not reuse `key` here. As far as I can tell, there's basically three cases:

- user provides a name: `'{}'.format(name)`
- user doesn't provide a name, regular build: `'{} #{}'.format(os.environ['BUILD_JOB_NAME'], os.environ['BUILD_NAME'])`

If I understand you correctly, you now are in a situation where you have two `key`s, but only one `BUILD_JOB_NAME`, right? I'm assuming this happens when you have two `put`s in the same job?

If so, I can see the need to use `key`, but we should probably do something about the automatically generated key for one-off builds.
- user doesn't provide a name, one-off build: `'One-off build #{}'.format(os.environ['BUILD_ID'])`